syntax = "proto3";
package Core.Engine.Network;

message BaseReq
{
	int32 protoId = 1;
	int32 requestId = 2;  //rpc 用
	bytes data = 3;       //proto 对应的二进制
}

message BaseRsp
{
	int32 protoId = 1;
	ErrorCode errorCode = 2;			//如果是rsp 协议 使用
 	int32 responseId = 3;		//如果是rsp 协议 使用
	int64 notifySeqId = 4;		//rsp ntf 均使用
	bytes data = 5;
}

//后续根据系统 不停增加
enum ErrorCode
{
	Err_OK = 0;   //0 表示 正确
	Err_Default = 1; // 默认错误
	Err_GmCommondReq = 2; 
	Err_CannotFindPlayer = 3; // 找不到player
	Err_InvalidCommand = 4; // command 频率太快, 后续优化
	Err_PlayerJoinReq = 5; 
	Err_ManageCastleReq = 6;
	Err_LevelUpWrongCastle = 7;
	Err_PendingItemReq = 8;
	Err_BagReq = 9;
	Err_EditSquadReq = 10;
	Err_TeamDefeat = 11; // 队伍战败时收到command
	Err_OrderSquadReq = 12;
	Err_BattlePlayerCommandReq = 13;
	Err_SellItemNotEnough = 14;
	Err_SellItemWrongItemId = 15;
	Err_LevelUpFullCastle = 16;
	Err_LevelUpCastleWrongTeam = 17;
	Err_LevelUpCastleResNotEnough = 18;
	Err_MoveCrystalTargetWrongTargetIdx = 19;
	Err_MoveCrystalWrongTeam = 20;
	Err_MoveCrystalNotFindCrystal = 21;
	Err_LordSelectWrongGameState = 22;
	Err_LordSelectWrongLordId = 23;
	Err_LordSelectWrongLockFlag = 24;
	Err_BuyPendingItemSoldOut = 25;
	Err_BuyPendingItemNotEnough = 26;
	Err_RefreshPendingItemNotEnough = 27;
	Err_AmbushHasPath = 28;
	Err_AmbushWrongState = 29;
	Err_AmbushWrongAp = 30;
	Err_AmbushFightNoAmbushPlayer = 31;
	Err_AmbushFightNoAmbushedPlayer = 32;
	Err_AmbushFightWrongAmbushState = 33;
	Err_AmbushFightWrongAmbushedState = 34;
	Err_AmbushFightWrongGridIdx = 35;
	Err_AttackEntityWrongAttackerState = 36;
	Err_AttackEntityWrongDefenderState = 37;
	Err_AttackEntityWrongAttackerPos = 38;
	Err_AttackEntityNotNeighbour = 39;
	Err_AttackEntityRepeatJoin = 40;
	Err_AttackEntityCannotJoin = 41;
	Err_AttackEntityAmbush = 42;
	Err_BornSquadWrongGridIdx = 43;
	Err_BornSquadHasEntity = 44;
	Err_BornSquadNotCrystalBelong = 45;
	Err_BornSquadHasBattle = 46;
	Err_BornSquadWrongState = 47;
	Err_CardLevelNotExist = 48;
	Err_CardLevelNotEnough = 49;
	Err_CardLevelFull = 50;
	Err_EconomyLevelNotExist = 51;
	Err_EconomyLevelNotEnough = 52;
	Err_EconomyLevelFull = 53;
	Err_JoinFightWrongType = 54;
	Err_JoinFightWrongDefendIdx = 55;
	Err_JoinFightWrongPos = 56;
	Err_JoinFightNotNeighbour = 57;
	Err_JoinFightWrongState = 58;
	Err_JoinFightNotBattle = 59;
	Err_JoinFightRepeatJoin = 60;
	Err_LevelUpEntityWrongType = 61;
	Err_LevelUpEntityFull = 62;
	Err_LevelUpEntityNotEnough = 63;
	Err_LoadingCancelWrongState = 64;
	Err_MoveAttackControlBuff = 65;
	Err_MoveAttackWrongState = 66;
	Err_MoveAttackWrongGridIdx = 67;
	Err_MoveAttackWrongPos = 68;
	Err_MoveAttackAmbush = 69;
	Err_MoveAttackRepeatJoin = 70;
	Err_MoveAttackWrongPath = 71;
	Err_MoveAttackRpNotEnough = 72;
	Err_MovePawnOutOfTerritory = 73;
	Err_MovePawnWrongType = 74;
	Err_MovePawnOnlyOne = 75;
	Err_MovePawnWrongPlayer = 76;
	Err_MovePawnPopularityNotEnough = 77;
	Err_ReRollControlBuff = 78;
	Err_ReRollWrongState = 79;
	Err_ReRollHasPath = 80;
	Err_ReRollApNotEnough = 81;
	Err_ReviveSquadWrongState = 82;
	Err_ReviveSquadTimeBefore = 83;
	Err_ReviveSquadNotEnough = 84;
	Err_ScoutWrongState = 85;
	Err_ScoutHasPath = 86;
	Err_ScoutWrongAp = 87;
	Err_ScoutWrongPos = 88;
	Err_ScoutFar = 89;
	Err_ScoutWrongTarget = 90;
	Err_TeleportWrongAp = 91;
	Err_TeleportWrongPos = 92;
	Err_TeleportWrongState = 93;
	Err_TeleportHasPath = 94;
	Err_TeleportNotInCastle = 95;
	Err_TeleportWrongTarget = 96;
	Err_TeleportInCd = 97;
	Err_UsePortalWrongState = 98;
	Err_UsePortalControlBuff = 99;
	Err_UsePortalHasPath = 100;
	Err_UsePortalWrongPos = 101;
	Err_UsePortalNotExist = 102;
	Err_SkillNotExist = 103;
	Err_SkillWrongDataId = 104;
	Err_SkillHasPath = 105;
	Err_SkillControlBuff = 106;
	Err_SkillWrongState = 107;
	Err_SkillNotEnough = 108;
	Err_SkillInCd = 109;
	Err_SkillUseOut = 110;
	Err_ButcherWrongParam = 111;
	Err_ButcherIllegalTarget = 112;
	Err_ButcherWrongTargetState = 113;
	Err_ButcherWrongTargetPos = 114;
	Err_ButcherWrongOriginIdx = 115;
	Err_ButcherTargetInCastle = 116;
	Err_ButcherOutOfRange = 117;
	Err_ButcherNoStayPos = 118;
	Err_RewindWrongParam = 119;
	Err_RewindUnableSummons = 120;
	Err_GeneralWrongParam = 121;
	Err_GeneralWrongPawnType = 122;
	Err_GeneralFullPawnLevel = 123;
	Err_GeneralWrongPawnLevel = 124;
	Err_ShadowWrongParam = 125;
	Err_ShadowUnableSummons = 126;
	Err_ShadowNoBornGrid = 127;
	Err_ShadowNoTargetGrid = 128;
	Err_ShadowOutOfRange = 129;
	Err_HackerWrongParam = 130;
	Err_HackerWrongGridIdx = 131;
	Err_HackerWrongVisionGrid = 132;
	Err_HackerWrongTeleportGrid = 133;
	Err_HackerWrongOutOfRange1 = 134;
	Err_HackerWrongOutOfRange2 = 135;
	Err_HackerNoCastle = 136;
	Err_HackerHasOther = 137;
	Err_HackerInBattle = 138;
	Err_IceWallWrongParam = 139;
	Err_IceWallWrongTargetIdx = 140;
	Err_IceWallUnablePut = 141;
	Err_PortalWrongParam = 142;
	Err_PortalWrongTargetGridIdx = 143;
	Err_PortalWrongPos = 144;
	Err_PortalHasCastle = 145;
	Err_PortalHasOther = 146;
	Err_PortalOutOfRange = 147;
	Err_PotentialWrongParam = 148;
	Err_PotentialWrongTargetType = 149;
	Err_PotentialFullAp = 150;
	Err_PotentialWrongTargetState = 151;
	Err_PotentialOutOfRange = 152;
	Err_SacrificeWrongParam = 153;
	Err_SacrificeWrongState = 154;
	Err_SacrificeWrongTargetType = 155;
	Err_SacrificeOutOfRange = 156;
	Err_TerrorWrongParam = 157;
	Err_TerrorWrongTargetType = 158;
	Err_TerrorWrongTargetState = 159;
	Err_TerrorOutOfRange = 160;
	Err_SurrenderNotStart = 161;
	Err_SurrenderIsInterval = 162;
	Err_SurrenderNotEnough = 163;
	Err_TechLevelFull = 164;
	Err_PlayerTeleportWrongState = 165;
	Err_PlayerTeleportHasPath = 166;
	Err_PlayerTeleportHasBuff = 167;
	Err_PlayerTeleportHasEntity = 168;
	Err_PlayerTeleportNotTerritory = 169;
	Err_PlayerTeleportWrongTerritory = 170;
	Err_PlayerTeleportWrongGridIdx = 171;
	Err_PlayerTeleportInCd = 172;
	Err_PlayerTeleportHasBattle = 173;
	Err_PlayerTeleportApNotEnough = 174;
	Err_PlayerTeleportWrongGrid = 175;
	Err_GeneralWrongPawnRainbow = 176;
	Err_ChronoAssistanceWrongTarget = 177;
	Err_ChronoAssistanceWrongParam = 178;
	Err_ChronoAssistanceWrongState = 179;
	Err_IceWallOutOfRange = 180;
	Err_IceWallWrongState = 181;
	Err_IceWallOutOfMap = 182;
	Err_SellItemWrongRainbowCard = 183;
	Err_RadishWrongPosition = 184;
	Err_SwitchRainbowSame = 185;
	Err_SkillAimPosOutOfRange = 186;
	Err_CollectRadishWrongState = 187;
	Err_GrowRadishWrongState = 188;
	Err_AmbushCheckFail = 189;
	Err_SkillChargeNotEnoughHp = 190;
	Err_SkillScoutWrongParam = 191;
	Err_SkillScoutWrongGridIdx = 192;
	Err_SkillGarrisonHasCastle = 193;
	Err_SkillGarrisonHasBuff = 194;
	Err_UrbanRelease = 195;
	Err_SkillNotHasPrepare = 196;
	Err_SkillSoulLink = 197;
	Err_SkillYoNeSoul = 198;
	
	Err_CommonParamError = 300;
	Err_CommonOutOfMap = 301;
	Err_CommonHasCastle = 302;
	Err_CommonGridFilled = 303;
	Err_CommonOutOfRange = 304;
	Err_GunCarrierWrongParam = 305;
	Err_GunCarrierUnableSummons = 306;
	Err_GunCarrierNoTargetGrid = 307;
}